### PR TITLE
Use RTD v2

### LINKED ISSUE #429: No such file or directory: readthedocs.yml
I'm creating a new package from the template using cookiecutter. If I answer 'n' to the question `use_read_the_docs [y]:`, the package creation fails with 

```
Traceback (most recent call last):
  File "/tmp/tmp7k_e0rii.py", line 48, in <module>
    remove_file('readthedocs.yml')
  File "/tmp/tmp7k_e0rii.py", line 9, in remove_file
    os.remove(os.path.join(PROJECT_DIRECTORY, filepath))
FileNotFoundError: [Errno 2] No such file or directory: '/home/mbachett/software/IXPE/ixpehealth/readthedocs.yml'
ERROR: Stopping generation because post_gen_project hook script didn't exit successfully

```

### LINKED ISSUE #431: MNT: Use ReadTheDocs v2 YAML and not conda
1. Upgrade from v1 to v2 ReadTheDocs YAML specifications. See https://docs.readthedocs.io/en/stable/config-file/v2.html
2. Stop using `conda` to set up the environment. Use `pip` all around for faster build time and more predictable outcome.

### PR DESCRIPTION
**Pull request by pllim**
Fix #429 

Fix #431

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -1,4 +1,20 @@
...
 
+sphinx:
+  builder: html
+  configuration: docs/conf.py
+  fail_on_warning: true
+
+# Set the version of Python and requirements required to build your docs
 python:
-   version: 3.5
+  version: 3.7
```
Comment (pllim) [inline .readthedocs.yml]: No reason to  not use Python 3.7?

File: hooks/post_gen_project.py [hunk 2]
```diff
@@ -44,8 +44,8 @@ def process_licence(licence_name):
         remove_file('.travis.yml')
 
     if '{{ cookiecutter.use_read_the_docs }}' != 'y':
-        remove_file('.rtd-environment.yml')
-        remove_file('readthedocs.yml')
+        remove_file('rtd-pip-requirements')
+        remove_file('.readthedocs.yml')
```
Comment (pllim) [inline hooks/post_gen_project.py]: Here is the typo fix for #429.

File: {{ cookiecutter.package_name }}/.readthedocs.yml [hunk 3]
```diff
@@ -1,5 +1,25 @@
...
+version: 2
 
+# Set "fail_on_warning: true" for RTD to fail on warnings.
+sphinx:
+  builder: html
+  configuration: docs/conf.py
+  fail_on_warning: false
+
+# Set the version of Python and requirements required to build your docs
+# NOTE: This does not follow {{ cookiecutter.minimum_python_version }}
```
Comment (pllim) [inline {{ cookiecutter.package_name }}/.readthedocs.yml]: I am not sure if one should build doc from *minimum* Python version, so I just hardcode it to Python 3.7.

File: .readthedocs.yml [hunk 4]
```diff
@@ -1,4 +1,20 @@
-requirements_file: requirements.txt
+# Read the Docs configuration file
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+version: 2
 
+sphinx:
```
Comment (bsipocz) [inline .readthedocs.yml]: we don't seem to have this section in any of our production config, but I suppose it won't hurt to list these defaults

File: .readthedocs.yml [hunk 5]
```diff
@@ -1,4 +1,20 @@
...
+  builder: html
+  configuration: docs/conf.py
+  fail_on_warning: true
+
+# Set the version of Python and requirements required to build your docs
 python:
-   version: 3.5
+  version: 3.7
+  system_packages: true
```
Comment (bsipocz) [inline .readthedocs.yml]: not sure whether this can cause any issues, I suppose it won't.

**Comment by bsipocz** [comment]
This looks good to me. The comments are rhetorical ones really, I suppose I'm happy to merge this as is.

File: {{ cookiecutter.package_name }}/.readthedocs.yml (hunk 3 above)
Comment (bsipocz) [reply {{ cookiecutter.package_name }}/.readthedocs.yml]: well, this is just a template, maintainers are expected to look under the hood, at least a little bit, and then your comment clears up the situation.

**Comment by astrofrog** [comment]
What if we declared an extras requires for docs in setup.cfg and put Sphinx-astropy there, then used that in the RTD config? Would be more consistent with the core package than using a rtd requirements file?

**Comment by astrofrog** [comment]
Just to be clear I’m ok with this being merged and happy to do a follow up PR

File: .readthedocs.yml (hunk 4 above)
Comment (pllim) [reply .readthedocs.yml]: Some of these are currently set in the RTD admin panel, but I think RTD devs wanna move away from that and define everything in the YAML eventually. Also being explicit doesn't hurt, as you said.

File: .readthedocs.yml (hunk 5 above)
Comment (pllim) [reply .readthedocs.yml]: I am actually not sure what gets installed as "system packages" but I have that option for all my projects and it doesn't seem to hurt. :woman_shrugging:

**Comment by pllim** [comment]
> Would be more consistent with the core package than using a rtd requirements file?

The core still uses a RTD requirements file on top of the `setup.cfg` to pin `matplotlib`. :grimacing: 

https://github.com/astropy/astropy/blob/4c46eb6b09c65b6e66db7d1a8a4b2a40224482bc/docs/rtd_requirements.txt#L1

I agree that using `setup.cfg` is a separate PR.

File: .readthedocs.yml (hunk 5 above)
Comment (bsipocz) [reply .readthedocs.yml]: OK, that's fine then. I guess  it's also the default, as in some weird cases we run into some version issues with their system installed packages (as I recall all could be traced back to some real issues with the packages/dependencies).

**Comment by pllim** [comment]
Feel free to merge if you are okay with this. I try not to merge my own PRs. :wink:

File: .readthedocs.yml [hunk 6]
```diff
@@ -1,4 +1,20 @@
...
+  install:
+    - requirements: requirements.txt
+    - method: setuptools
```
Comment (Cadair) [inline .readthedocs.yml]: why `setuptools` and not `pip`?

File: {{ cookiecutter.package_name }}/.readthedocs.yml [hunk 7]
```diff
@@ -1,5 +1,25 @@
...
 python:
-   setup_py_install: true
+  version: 3.7
+  system_packages: true
+  install:
+    - requirements: rtd-pip-requirements
+    - method: setuptools
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/.readthedocs.yml]: Why setuptools and not pip?

File: {{ cookiecutter.package_name }}/rtd-pip-requirements [hunk 8]
```diff
@@ -0,0 +1,5 @@
+Cython
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/rtd-pip-requirements]: Why don't we move this file to `docs/requirements.txt` or `docs/rtd-requirements.txt`? I feel like we don't need another file in the root of the repo.

File: .readthedocs.yml (hunk 6 above)
Comment (pllim) [reply .readthedocs.yml]: To run `python setup.py install`. I think @astrofrog already brought it up above and said the `pip` thing can be a separate PR.

File: {{ cookiecutter.package_name }}/.readthedocs.yml (hunk 7 above)
Comment (pllim) [reply {{ cookiecutter.package_name }}/.readthedocs.yml]: Same answer as above.

File: {{ cookiecutter.package_name }}/rtd-pip-requirements (hunk 8 above)
Comment (pllim) [reply {{ cookiecutter.package_name }}/rtd-pip-requirements]: I can move it as long as there is a consensus. I feel people have been flip-flopping where that file should go throughout the years and we move it back and forth.

Comment (bsipocz) [reply {{ cookiecutter.package_name }}/rtd-pip-requirements]: I don't have a preference, but picking one and sticking to it in all the repos would be nice. So far astropy and photutils have `docs/rtd_requirements.txt`

Comment (pllim) [reply {{ cookiecutter.package_name }}/rtd-pip-requirements]: Okay, I moved and renamed it. Also squashed commits.

**Comment by Cadair** [comment]
On the `[docs]` extras require point, we found in SunPy that we needed to be able to specify things specific to RTD before, which made extras require not a perfect solution. I don't know if that's a concern we want to carry into the package template though.

**Comment by pllim** [comment]
@Cadair , I am not using `[docs]` thingy here... am I?

**Comment by Cadair** [comment]
No, I was responding to @astrofrog's comments.

**Comment by pllim** [comment]
Superseded by #438

**Pull request closed without merging**